wpilibc/src/main/native/cpp/ErrorBase.cpp

```diff
@@ -23,6 +23,10 @@ using namespace frc;
 std::mutex ErrorBase::_globalErrorMutex;
 Error ErrorBase::_globalError;
 
+ErrorBase::ErrorBase() {
```

Comment:
Why does calling `HAL_Initialize()` in `#define START_ROBOT_CLASS(_ClassName_)` not work? Is there a case that doesn't use `START_ROBOT_CLASS`'s definition of `main()`?

Reply:
See #663 for why that does not work in all cases.

Reply:
OK yeah. Each device that uses the HAL should be calling `HAL_Initialize()` in its constructor (or in some device's case, ErrorBase).